db/db.go

```diff
@@ -0,0 +1,33 @@
+package db
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/smallstep/nosql"
+)
+
+// dbConfig represents the JSON attributes used for configuration. At
+// this moment only fields for NewRelic are supported.
+type dbConfig struct {
+	Type string `json:"type"`
+	Path string `json:"path"`
+}
+
+// New returns a new database client that implements the nosql.DB interface.
+func New(raw json.RawMessage) (nosql.DB, error) {
```

Comment:
Change it to something like `func New(config Config) (DB, error)` where Config is the configuration above and DB is a new type that wraps all the functionality used ie: `func (db *DB) IsRevoked(sn string) (bool, error)`